woocommerce/woocommerce
# Add IframeEditor component to product editor

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR adds in an `IframeEditor` for use in pages to allow us to edit specific sections, such as the product description, within a nested editor.

Note that some follow-ups will be needed and we'll need some design input moving forward (cc @jarekmorawski):

* Back/exit button
* Callback functions for `onChange`, `onClose`, etc.
* Implementation of this block within the description
* Updates to the design of this component

<img width="832" alt="Screen Shot 2023-04-04 at 5 25 23 PM" src="https://user-images.githubusercontent.com/10561050/229951032-8b29c58d-83d8-490c-af0d-b8b152521b78.png">

Closes #37266  .



### How to test the changes in this Pull Request:

1. Either check out the branch `add/37266-test` OR add the `IframeEditor` component manually to a page by importing it from `@woocommerce/product-editor`.
2. Navigate to Tools -> WCA Test Helper -> Features and enable the new product blocks editing experience
3. Navigate to Products -> Add new (or where you manually added the new component)
4. Check that blocks can be added and blocks styles exist
5. Make sure you can resize the editor as needed
